### PR TITLE
Prefix seed_everything log messages with rank info

### LINKED ISSUE #13290: Prefix `seed_everything` log messages with rank info
## 🚀 Feature

The message here is printed per rank on purpose https://github.com/PyTorchLightning/pytorch-lightning/blob/bfa8b7be2d99b980afa62f5cb0433326bcfd2ef0/pytorch_lightning/utilities/seed.py#L69-L71 but this confuses users who are not aware of the difference

### Pitch

Prefix the message with the rank just as we do for `EarlyStopping` https://github.com/PyTorchLightning/pytorch-lightning/blob/bfa8b7be2d99b980afa62f5cb0433326bcfd2ef0/pytorch_lightning/callbacks/early_stopping.py#L256-L261

the above staticmethod should be pulled into a common utility function.
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

cc @borda @awaelchli

### PR DESCRIPTION
**Pull request by MrShevan**
## What does this PR do?

Fixes #13290 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/utilities/seed.py [hunk 1]
```diff
@@ -153,3 +151,14 @@ def isolate_rng() -> Generator[None, None, None]:
...
     _set_rng_states(states)
+
+
+def _log_info(message: str) -> None:
+    """Check global rank if the rank is greater than zero, add a prefix."""
+    rank = _get_rank()
+
+    if rank > 0:
+        log.info(f"[rank: {rank}] {message}")
+        return
```
Comment (awaelchli) [inline src/pytorch_lightning/utilities/seed.py:162]: Why are we excluding rank 0?

Comment (MrShevan) [reply src/pytorch_lightning/utilities/seed.py:162]: Hi, thanks for watching! I read your comment to [issue](https://github.com/Lightning-AI/lightning/issues/13290#issuecomment-1159567609) that in some cases, _get_rank() might the prefix will falsely display rank 0 for all messages. I thought that maybe in such cases we can not change the logging behavior.

Do you think it's better to always specify rank in logging? I can do so.

Comment (awaelchli) [reply src/pytorch_lightning/utilities/seed.py:162]: Yes, on some custom clusters `_get_rank()` will return 0 at the time of calling `seed_everything()`, because the true rank information is not yet available. You can't avoid this, even when you do a check for `rank > 0`, it won't be true and so can't address this problem. It's not a huge issue, and we can't really do anything about it. 

> Do you think it's better to always specify rank in logging? I can do so

yes

Comment (MrShevan) [reply src/pytorch_lightning/utilities/seed.py:162]: Got it! Thanks for the explanation, I will make changes

**Comment by MrShevan** [comment]
@carmocca, hi! Could you review this?

File: src/pytorch_lightning/CHANGELOG.md [hunk 2]
```diff
@@ -103,6 +103,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added support for DDP Fork ([#13405](https://github.com/Lightning-AI/lightning/pull/13405))
 - Added support for async checkpointing ([#13658](https://github.com/Lightning-AI/lightning/pull/13658))
 - Added support for HPU Device stats monitor ([#13819](https://github.com/Lightning-AI/lightning/pull/13819))
+- Added prefix to log message in `seed_everything` with rank info ([#13290](https://github.com/Lightning-AI/lightning/issues/13290))
```
Comment (awaelchli) [inline src/pytorch_lightning/CHANGELOG.md:106]: Needs to move up to the new section :)

File: src/pytorch_lightning/utilities/seed.py [hunk 3]
```diff
@@ -153,3 +151,8 @@ def isolate_rng() -> Generator[None, None, None]:
     states = _collect_rng_states()
     yield
     _set_rng_states(states)
+
+
+def _log_info(message: str) -> None:
```
Comment (awaelchli) [inline src/pytorch_lightning/utilities/seed.py:156]: We don't need a helper function here, right? log.info can be called directly in the seed function above IMO

Comment (MrShevan) [reply src/pytorch_lightning/utilities/seed.py:156]: I used this trick for easy testing. Without helper function I have to know SEED value to generate expected message to log.info and if I use None as SEED in `seed_everything` function it generates randomly. So, I used the helper function to check logic of adding rank info despite logging message.

File: src/pytorch_lightning/CHANGELOG.md (hunk 2 above)
Comment (MrShevan) [reply src/pytorch_lightning/CHANGELOG.md:106]: You mean section 1.8.0 ?

File: src/pytorch_lightning/utilities/seed.py (hunk 3 above)
Comment (awaelchli) [reply src/pytorch_lightning/utilities/seed.py]: If you want to add a test for that, it should be easy to mock this. You can mock directly the log.info method and intercept the message that goes in.

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/CHANGELOG.md (hunk 2 above)
Comment (awaelchli) [reply src/pytorch_lightning/CHANGELOG.md]: Yes

**Review by otaj — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: tests/tests_pytorch/utilities/test_seed.py [hunk 4]
```diff
@@ -96,3 +98,19 @@ def test_isolate_rng():
     with isolate_rng():
         generated = [random.random() for _ in range(3)]
     assert random.random() == generated[0]
+
+
+@mock.patch("pytorch_lightning.utilities.seed.log.info")
+@pytest.mark.parametrize("env_vars", [{"RANK": "0"}, {"RANK": "1"}, {"RANK": "4"}])
+def test_seed_everything_log_info(log_mock: MagicMock, env_vars: Mapping[str, str]):
+    """Test that log message prefix with correct rank info."""
+    with mock.patch.dict(os.environ, env_vars):
```
Comment (rohitgr7) [inline tests/tests_pytorch/utilities/test_seed.py:107]:
```suggestion
    with mock.patch.dict(os.environ, env_vars, clear=True):
```

**Review by rohitgr7 — APPROVED** [review]

File: src/pytorch_lightning/CHANGELOG.md [hunk 5]
```diff
@@ -116,6 +116,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added support for async checkpointing ([#13658](https://github.com/Lightning-AI/lightning/pull/13658))
 - Added support for HPU Device stats monitor ([#13819](https://github.com/Lightning-AI/lightning/pull/13819))
 
+
```
Comment (carmocca) [inline src/pytorch_lightning/CHANGELOG.md:119]:
```suggestion
```

**Review by carmocca — APPROVED** [review]
